### PR TITLE
Port share by mail settings to vue

### PR DESCRIPTION
**Pull request by CarlSchwan**

### CODE REVIEW

**Comment by CarlSchwan** [comment]
I'm not sure why npm is doing so much dist changes :(

**Review by skjnldsv — APPROVED** [review]

File: apps/sharebymail/src/components/AdminSettings.vue [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		},
+		async handleResponse({ status, errorMessage, error }) {
+			if (status !== 'ok') {
+				showError(errorMessage)
+				console.error(errorMessage, error)
+			}
+		},
+	}
+}
+</script>e
```
Comment (CarlSchwan) [inline apps/sharebymail/src/components/AdminSettings.vue:95]:
```suggestion
</script>
```

File: apps/sharebymail/src/components/AdminSettings.vue [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	-
+	- You should have received a copy of the GNU Affero General Public License
+	- along with this program. If not, see <http://www.gnu.org/licenses/>.
+	-
+-->
+
+<template>
+	<SettingsSection :title="t('sharebymail', 'Share by mail')"
+		:description="t('sharebymail', 'Allows users to share a personalized link to a file or folder by putting in an email address.')"
+		:doc-url="sharingFederatedDocUrl">
```
Comment (artonge) [inline apps/sharebymail/src/components/AdminSettings.vue:26]: `sharingFederatedDocUrl` seems to be not declared anywhere.

File: apps/sharebymail/src/components/AdminSettings.vue [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+export default {
+	name: 'AdminSettings',
+	components: {
+		CheckboxRadioSwitch,
+		SettingsSection,
+	},
+	data() {
+		return {
+			sendPasswordMail,
+			replyToInitiator,
```
Comment (artonge) [inline apps/sharebymail/src/components/AdminSettings.vue:62]: This also work, and you can remove the declaration of the intermediate variables above.

```suggestion
			sendPasswordMail: loadState('sharebymail', 'sendPasswordMail'),
			replyToInitiator: loadState('sharebymail', 'replyToInitiator'),
```

**Review by artonge — CHANGES_REQUESTED** [review]

**Review by artonge — APPROVED** [review]

**Pull request merged**